spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AbstractAmqpOutboundEndpoint.java

```diff
@@ -284,6 +313,13 @@ protected final void doInit() {
 			Assert.state(this.confirmNackChannel == null || nullChannel != null,
 					"A 'confirmCorrelationExpression' is required when specifying a 'confirmNackChannel'");
 		}
+		if (this.delayExpression != null) {
+			this.delayGenerator = new ExpressionEvaluatingMessageProcessor<Integer>(this.delayExpression,
```

Comment:
I don't see reason in this extra object, but let it be: it is existing logic in this component.